apps/ws/src/socket/usecases/external-services-route/external-services-route.usecase.ts

```diff
@@ -0,0 +1,84 @@
+import { Injectable } from '@nestjs/common';
+
+import { MessageRepository } from '@novu/dal';
+import { ChannelTypeEnum } from '@novu/shared';
+import { BullMqService } from '@novu/application-generic';
+
+import { ExternalServicesRouteCommand } from './external-services-route.command';
+import { WSGateway } from '../../ws.gateway';
+
+@Injectable()
+export class ExternalServicesRoute {
+  public readonly bullMqService: BullMqService;
+
+  constructor(private wsGateway: WSGateway, private messageRepository: MessageRepository) {
+    this.bullMqService = new BullMqService();
+  }
+  public async execute(command: ExternalServicesRouteCommand) {
+    if (command.event === 'unseen_count_changed') {
+      await this.sendUnseenCountChange(command);
+
+      return;
+    }
+
+    if (command.event === 'unread_count_changed') {
+      await this.sendUnreadCountChange(command);
+
+      return;
+    }
+
+    await this.wsGateway.sendMessage(command.userId, command.event, command.payload);
+  }
+
+  private async sendUnreadCountChange(command: ExternalServicesRouteCommand) {
+    if (!command._environmentId) {
+      return;
+    }
+
+    const connection = await this.connectionExist(command);
+
+    if (!connection) {
+      return;
+    }
```

Comment:
i refactored this block into a separate function so i could reuse it but in the end, it did not look better.